osquery/core/init.cpp

```diff
@@ -97,13 +101,20 @@ enum {
 namespace {
 extern "C" {
 static inline bool hasWorkerVariable() {
-  return (getenv("OSQUERY_WORKER") != nullptr);
+  auto value = ::osquery::getEnvVar("OSQUERY_WORKER");
+  if (value) {
```

Comment:
It should work fine right? This seems to be how they use it in the example in their doc http://www.boost.org/doc/libs/1_60_0/libs/optional/doc/html/index.html